data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/affect_distant_active_with_second_ability.cfg

```diff
@@ -0,0 +1,119 @@
+    [event]
+        name=start
+        [unit]
+            id=alex
+            name=_"Alex"
+            x,y=1,1
+            type=Elvish Hero
+            side=1
+        [/unit]
+        [unit]
+            id=bobby
+            name=_"Bobby"
+            x,y=1,5
+            type=Elvish Hero
+            side=1
+        [/unit]
+        [unit]
+            id=charlie
+            name=_"Charlie"
+            x,y=1,12
+            type=Elvish Hero
+            side=1
+        [/unit]
+
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [leadership]
+                        id=specialY
+                        name=_ "specialY"
+                        description=_ "specialY is active if one unit is within 7 hex radius"
+                        value=25
+                        affect_self=no
+                        affect_allies=yes
+                        affect_enemies=yes
+                        [affect_distant]
+                            radius=7
+                        [/affect_distant]
+                    [/leadership]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=charlie
+            [/filter]
+        [/object]
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [damage]
+                        id=specialX
+                        name=_ "specialX"
+                        description=_ "specialX is active if within radius of specialY ability"
+                        value=100
+                        apply_to=self
+                        affect_self=no
+                        affect_allies=yes
+                        affect_enemies=yes
+                        [filter]
+                            {FILTER}
+                        [/filter]
+                        [affect_distant]
+                            radius=5
+                        [/affect_distant]
+                    [/damage]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=bobby
+            [/filter]
+        [/object]
+
+        {ASSERT (
+            [have_unit]
+                id=alex
+                ability_id_active=specialX
+            [/have_unit]
+        )}
+
+        {SUCCEED}
+    [/event]
+#enddef
+
+#####
+# API(s) being tested: ability[affect_distant]radius=
+##
+# Actions:
+# Give bobby an ability specialX, which is affect alex if he in 5 hexes of distance of bobby and active if an unit with [leadership] abilitie is within a radius of 7 of bobby.
+# Give charlie an [leadership] ability, which is affect bobby if he in 7 hexes of distance of charlie.
+# Test whether the ability is active.
+##
+# Expected end state:
+# specialX should be active and affect alex.
+#####
+{GENERIC_UNIT_TEST "affect_distant_active_with_second_ability_type" (
```

Comment:
I done a bad choisi of the map who is too litle. I retested manually and it work when all three unit can be placed.  I fix that tomorrow

Reply:
@Pentarctagon test validated